### PR TITLE
Implement workflow libraryrun api

### PR DESCRIPTION
**Pull request by williamputraintan**
APIs created for workflow-libraryrun link:
- /libraryrun/by_workflow/
  Return libraryrun based on workflow keywords in parameter:
    - end_status
    - type

- /workflows/by_library_id/
  Return workflows based on library_id in parameter

### CODE REVIEW

File: data_portal/models.py [hunk 1]
```diff
@@ -865,6 +866,16 @@ def get_by_keyword(self, **kwargs) -> QuerySet:
 
         return qs
 
+    def get_workflow_by_library_id(self, **kwargs):
+        qs: QuerySet = self.all()
```
Comment (reisingerf) [inline data_portal/models.py:870]: This is returning all workflows if a library id is not found?

File: data_portal/models.py [hunk 2]
```diff
@@ -1068,6 +1079,20 @@ def get_by_keyword(self, **kwargs) -> QuerySet:
...
         return qs
 
+    def get_library_by_workflow_keyword(self, **kwargs):
+        qs: QuerySet = self.all()
+
+        type_name = kwargs.get('type_name', None)
+        if type_name:
+            qs = qs.filter(workflows__type_name__iexact=type_name)
+
+        status_list_string = kwargs.get('end_status', None)
```
Comment (reisingerf) [inline data_portal/models.py:1089]: This is supporting a multi-value string for the `end_status`?
What's the use case?

**Review by reisingerf — COMMENTED** [review]
Not sure how the querying works, but does it not return too many results when the provided argument(s) don't match expectation?

File: data_portal/models.py (hunk 2 above)
Comment (williamputraintan) [reply data_portal/models.py:1089]: Thinking of using it as the status filter bar as the client could select multiple statuses

image

Comment (reisingerf) [reply data_portal/models.py:1087]: Perhaps for very specific requests we could consider a POST with a query payload?

Comment (victorskl) [reply data_portal/models.py:1087]: Yep; as discussed, let stick to single value `status=foo` for now. We will revisit when use case stronger for multi-values.

**Pull request merged**